scripts/src/test-raffle.ts

```diff
@@ -0,0 +1,554 @@
+
+  const claim = new MsgExecuteContract(
+    wallet.key.accAddress("terra"),
+    airdropContract,
+    {
+      claim: {
+        signature: signature,
+        claim_amounts: claim_amounts_string,
+      },
+    },
+    {}
+  );
+  const tx = await wallet.createAndSignTx({
+    msgs: [claim],
+    chainID: "pisco-1",
+  });
+  console.log(tx);
+  console.log("----------------------------------");
+  const res = await terra.tx.broadcast(tx, "pisco-1");
+  console.log(res);
+  return res;
+}
+
+async function transferUnclaimedTokens(
+  wallet: Wallet,
+  airdropContract: string,
+  amount: number
+) {
+  const transferUnclaimed = new MsgExecuteContract(
+    wallet.key.accAddress("terra"),
+    airdropContract,
+    {
+      transfer_unclaimed_tokens: {
+        amount: amount.toString(),
+        recipient: wallet.key.accAddress("terra"),
+      },
+    },
+    {}
+  );
+  const tx = await wallet.createAndSignTx({
+    msgs: [transferUnclaimed],
+    chainID: "pisco-1",
+  });
+  console.log(tx);
+  console.log("----------------------------------");
+  const res = await terra.tx.broadcast(tx, "pisco-1");
+  console.log(res);
+  return res;
+}
+
+async function wait(ms: number) {
+  return new Promise((resolve) => {
+    setTimeout(resolve, ms);
+  });
+}
+
+async function waitUntil(s: number) {
+  const timeUntil = s * 1000;
+  console.log("timeUntil", timeUntil);
+  const waiting_time =
+    timeUntil - Date.now() > 0 ? timeUntil - Date.now() + 60000 : 60000;
+  // try to wait until RPC/LCD and feather.js is updated with a time in the future
+  // takes at least 1 minute to update properly
+  console.log("waiting_time", waiting_time);
+  return new Promise((resolve) => {
+    setTimeout(resolve, waiting_time);
+  });
+}
+
+const expectError = (message) => (err: Error) => {
+  if (err.message !== "Error not thrown") {
+    console.log(message);
+  } else {
+    throw new Error("Error not thrown");
+  }
+};
+const throwErr = (_: any) => {
+  throw new Error("Error not thrown");
+};
+
+testall();
+
+async function testall() {
+  console.log("UPLOADING CONTRACTS");
+  const factoryContract = await uploadContract(hallwallet).then(async (res) => {
+    await wait(6000);
+    console.log("INSTANTIATING FACTORY");
+    return instantiateFactory(hallwallet, res[0], res[1]);
+  });
+  await wait(6000);
+
+  // const factoryContract =
+  //   "terra1ydwlh3auwwhn7xl4fn5zaeqx7xktmd9kqp0la4da3zxd7t6frjws2j50st";
+
+  // console.log("TESTING MULTI TEST 1");
+  // await test1(factoryContract).catch((err) => {
+  //   console.log(err);
+  // });
+  // await wait(6000);
+
+  // console.log("TESTING SINGLE TEST 1");
+  // await single_test1(factoryContract).catch((err) => {
+  //   console.log(err);
+  // });
+  // await wait(6000);
+
+  // console.log("TESTING SINGLE TEST 2");
+  // await single_test2(factoryContract).catch((err) => {
+  //   console.log(err);
+  // });
+  // await wait(6000);
```

Comment:
whats the reason for commenting out these test cases? If they're unused i think we should just remove them to reduce clutter + this is a public repo lol